src/components/pin/PinInputForm.jsx

```diff
@@ -0,0 +1,75 @@
+"use client";
+import React, { useState, useEffect, useCallback } from "react";
+import {
+  MODES,
+  shuffleArray,
+  getInstructionText,
+  KEYS,
+  PIN_LENGTH,
+} from "./pinUtils";
+import PinInputDisplay from "./PinInputDisplay";
+import KeypadButton from "./KeypadButton";
+import { usePinLogic } from "../../hooks/usePinLogic";
+
+/**
+ *
+ * @param {string} mode "setup"(초기 비밀번호 등록) | "verify"(비밀번호로 본인인증)
+ * @param {string} storedPin db에 저장된 사용자의 비밀번호
+ * @param {function} onSuccess 비밀번호 검증 후 동작할 함수
+ * @returns
+ */
+const PinInputForm = ({ mode, storedPin = "", onSuccess }) => {
+  const [shuffledNumbers, setShuffledNumbers] = useState(() =>
+    shuffleArray([...Array(10).keys()].map(String)),
+  );
```

Comment:
이 부분에서 Uncaught Error: Hydration failed 에러가 발생하네요..!
next.js 는 ssr방식으로 렌더링을 합니다. 그리고 클라이언트 컴포넌트면 csr로 한번 더 렌더링 하게되는데, 이 때 ssr방식으로 렌더링 된 html과 csr로 렌더링하는 html이 다르게 되면 Hydration failed 이 발생합니다
아래와같이 초기 값은 shuffle되지 않은 배열로 해주고, 마운트된 후에 shuffle을 해주면 될 것 같네요!

```suggestion
  const [shuffledNumbers, setShuffledNumbers] = useState(() =>
    [...Array(10).keys()].map(String),
  );
```

[- 참고 자료](https://techblog.gccompany.co.kr/next-js-%EA%B0%9C%EB%B0%9C-%EC%A4%91-%EB%A7%88%EC%A3%BC%ED%95%9C-hydration-error%EC%99%80-%EC%9B%90%EC%9D%B8-a439ee2345aa)

Reply:
Hydration 방지를 위해 초기 배열은 shuffle을 하지 않고, 이후 useEffect를 이용하여 렌더링 된 이후에 shuffle이 되도록 수정해 보았습니다! 하지만 현재 React에서는 useEffect 내부에서 set 함수를 사용하지 않도록 권장하고 있더라구요 

```
> Error: Calling setState synchronously within an effect can trigger cascading renders
> 
> Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
> * Update external systems with the latest state from React.
> * Subscribe for updates from some external system, calling setState in a callback function when external state changes.
```

(이유는 렌더링 -> 커밋 -> Effect -> set 함수로 상태 변경 -> 렌더링이 종료되기 전에 다시 렌더링 ..  이런 식으로 하나의 렌더 사이클이 종료되기 전에 상태가 바뀌어서 이를 경고하고 있어요)

따라서 비동기 함수를 이용해서, 렌더 사이클이 종료된 후에 set함수를 이용해서 shuffle을 진행하도록 로직을 변경해두었습니다! 🙆🏻🙆🏻